### ISSUE TITLE
Bring in hillslope hydrology

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/CTSM#222]
Notes on bringing the `hillslope_hydrology` branch to master.

Items to do:
- [ ] Reconcile changes in SoilHydrologyMod: https://github.com/ESCOMP/ctsm/issues/222#issuecomment-360518010
- [ ] Review changes to make sure nothing was inadvertently deleted: https://github.com/ESCOMP/ctsm/issues/222#issuecomment-360518357

**Comment by billsacks** [comment]
Reconcile changes in SoilHydrologyMod:

On the `unified_land_model` branch, I'm incorporating most of the `hillslope_hydrology` changes in SoilHydrologyMod, including the main changes in LateralFlowHillslope. So for the most part, we can keep the version of SoilHydrologyMod from the `unified_land_model` branch, but see below for exceptions.

As of 9c3d3ce8 on the `hillslope_hydrology` branch, I have incorporated all changes except:

- I am NOT setting qdischarge (`qdischarge(c) = qflx_latflow_out_vol(c)`)

- I have NOT incorporated this change in SurfaceRunoff:

```fortran
!scs: only allow non-zero saturated areas in lowest hillslope column
      do fc = 1, num_hillslope
         c = filter_hillslopec(fc)
         if (col%cold(c) /= ispval) fsat(c) = 0._r8
      enddo
```

**Comment by billsacks** [comment]
Review changes to make sure nothing was inadvertently deleted:

In looking at some diffs on the hillslope branch, I noticed that it reverts some changes that were made on the CLM trunk, mistakenly. This problem is in the svn branch, and then has been carried over to the git branch, too.

The one issue I noticed is in clm_varctl.F90, where the hillslope_hydrology branch brings some variables back into clm_varctl.F90 that were deleted in the trunk: repartition_rain_snow and glcmec_downscale_longwave. I haven't looked to see if there are other problems, too.

From doing some analysis of the history, my best guess is that this problem arose due to plugging in SourceMods that were based on one CLM version into a branch that started from a different CLM version: The hillslope_hydrology branch started from clm4_5_14_r223, whereas the above variables were removed in r221.

A careful review should be done of all diffs in the hillslope_hydrology branch before it is brought to master.
